app/build.gradle

```diff
@@ -47,19 +46,19 @@ android {
 dependencies {
     // There is a better way to declare and structure dependencies.
     // However, that is a different topic.
-    def daggerVersion = '2.14'
-    def butterKnifeVersion = '8.8.0'
-    def supportVersion = '26.0.1'
-    def kotlinVersion = '1.2.0'
+    def daggerVersion = '2.14.1'
+    def butterKnifeVersion = '8.8.1'
+    def supportVersion = '27.1.0'
+    def kotlinVersion = '1.2.21'
 
     kapt "com.google.dagger:dagger-compiler:$daggerVersion"
     kapt "com.google.dagger:dagger-android-processor:$daggerVersion"
     kapt "com.jakewharton:butterknife-compiler:$butterKnifeVersion"
 
-    compile "com.google.dagger:dagger:$daggerVersion"
-    compile "com.google.dagger:dagger-android:$daggerVersion"
-    compile "com.google.dagger:dagger-android-support:$daggerVersion"
-    compile "com.jakewharton:butterknife:$butterKnifeVersion"
-    compile "com.android.support:appcompat-v7:$supportVersion"
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlinVersion"
+    implementation "com.google.dagger:dagger:$daggerVersion"
```

Comment:
Starting with Android Gradle plugin v3.0.0, the new [Java Library plugin configurations](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph) introduced in Gradle 3.4 (`implementation` instead of `compile` are among one of them) are used. 

> For more info, see https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html#new_configurations.